src/Agent/NewRelic/Agent/Core/BrowserMonitoring/BrowserMonitoringScriptMaker.cs

```diff
@@ -73,11 +74,13 @@ public string GetScript(IInternalTransaction transaction)
             if (Log.IsFinestEnabled)
                 transaction.LogFinest("RUM: TryGetBrowserTimingHeader success.");
 
+            var scriptNonce = nonce is null ? "" : $@" nonce=""{HttpUtility.HtmlAttributeEncode(nonce)}""";
+
             // The JavaScript variable NREUMQ stands for New Relic End User Metric "Q". This was the name before marketing renamed "EUM" to "RUM".
             // We can't change the name of the variable since: (a) we have to be consistent across agents, and (b) it has to be in sync with the rum.js file which is downloaded from NR servers.
             var javascriptAgentConfiguration = $"window.NREUM||(NREUM={{}});NREUM.info = {browserConfigurationData.ToJsonString()}";
             var javascriptAgent = _configurationService.Configuration.BrowserMonitoringJavaScriptAgent;
-            return $"<script type=\"text/javascript\">{javascriptAgentConfiguration}</script><script type=\"text/javascript\">{javascriptAgent}</script>";
+            return $"<script type=\"text/javascript\"{scriptNonce}>{javascriptAgentConfiguration}</script><script type=\"text/javascript\">{javascriptAgent}</script>";
```

Comment:
Looks like there's a second script that would also need a `nonce` to avoid CSP violations.
```suggestion
            return $"<script type=\"text/javascript\"{scriptNonce}>{javascriptAgentConfiguration}</script><script type=\"text/javascript\"{scriptNonce}>{javascriptAgent}</script>";
```

Or is there something I'm missing here?

Reply:
Nope, you're right; I missed that. Will need to update tests, too.